<a href="https://colab.research.google.com/github/raulrosapacheco/BrasileiraoAnalise/blob/main/AnaliseBrasileirao.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [5]:
!pip install -q datadotworld[pandas]

In [2]:
# Autenticando integração com data.world
# Token: eyJhbGciOiJIUzUxMiJ9.eyJzdWIiOiJweXRob246cmF1bHJvc2EiLCJpc3MiOiJjbGllbnQ6cHl0aG9uOmFnZW50OnJhdWxyb3NhOjplODdjNTAwNi0zYWRkLTQwNzUtYjRjNy0yNzIxN2U5ZjBiZmMiLCJpYXQiOjE2NzczNTUwODEsInJvbGUiOlsidXNlcl9hcGlfYWRtaW4iLCJ1c2VyX2FwaV9yZWFkIiwidXNlcl9hcGlfd3JpdGUiXSwiZ2VuZXJhbC1wdXJwb3NlIjp0cnVlLCJzYW1sIjp7fX0.IQJarGjUzPMMEtd2d9aqo3p7h10I4z3z2uaiMDUBYihtv-L9J2jmBXfBs9Sm-9om7RxRpO0_kOy6PwTfYSGoTg
!dw configure

API token (obtained at: https://data.world/settings/advanced): eyJhbGciOiJIUzUxMiJ9.eyJzdWIiOiJweXRob246cmF1bHJvc2EiLCJpc3MiOiJjbGllbnQ6cHl0aG9uOmFnZW50OnJhdWxyb3NhOjplODdjNTAwNi0zYWRkLTQwNzUtYjRjNy0yNzIxN2U5ZjBiZmMiLCJpYXQiOjE2NzczNTY3MzksInJvbGUiOlsidXNlcl9hcGlfYWRtaW4iLCJ1c2VyX2FwaV9yZWFkIiwidXNlcl9hcGlfd3JpdGUiXSwiZ2VuZXJhbC1wdXJwb3NlIjp0cnVlLCJzYW1sIjp7fX0.YfkhFsAdnbmRVK_DQ-u0jfVnUl8ppjRD1WXchZ6_kEdbr6E-b0YKsp5w6c6JOQmZ5q3s4FVI86J2ELhmjPRWVA


In [18]:
import datadotworld as dw
from datetime import datetime

In [4]:
# Carregando dataset
brasileirao = dw.load_dataset('raulrosa/brasileiro1522')

In [8]:
# Carregando dataframes
partidas = brasileirao.dataframes['partidas']
gols = brasileirao.dataframes['gols']
cartoes = brasileirao.dataframes['cartoes']
estatisticas = brasileirao.dataframes['estatisticas']

In [9]:
# Valores Ausentes: partidas
partidas.isnull().sum()

partida_id           0
rodada               0
data                 0
hora                 0
mandante             0
visitante            0
tecnico_mandante     2
tecnico_visitante    1
vencedor             0
arena                0
mandante_placar      0
visitante_placar     0
mandante_estado      0
visitante_estado     0
dtype: int64

In [12]:
valores_ausentes = partidas[partidas.isna().any(axis=1)]
valores_ausentes

partida_id  rodada       data      hora   mandante    visitante  \
263         5250      27 2015-09-20  11:02:00      Goias    Joinville   
516         5503      14 2016-07-10  15:59:00  Sao Paulo   America-MG   
1042        6029      29 2017-10-18  21:47:00       Avai  Botafogo-RJ   

            tecnico_mandante      tecnico_visitante   vencedor  \
263                      NaN     P. Lopes de Gusmão      Goias   
516                      NaN  S. de Oliveira Vieira  Sao Paulo   
1042  C. dos Santos Oliveira                    NaN          -   

                                                  arena  mandante_placar  \
263   Estádio do Governo do Estado de Goiás (Serra D...                3   
516                                             Morumbi                3   
1042                                          Ressacada                1   

      visitante_placar mandante_estado visitante_estado  
263                  0              GO               SC  
516                  0              SP               MG  
1042                 1              SC               RJ

Após pesquisas:
- Técnico do Goiás (partida_id = 5250): **Wanderley José Ferreira Filho** 
- Técnico do São Paulo (partida_id = 5503): **André Soares Jardine** 
- Técnico Botafogo-RJ (partida_id = 6029): **Emílio Carlos Tannus Faro** 

Por reprsentar poucos valores, será feito a imputação destes dados. 

In [29]:
partidas_filtro = partidas[partidas['tecnico_mandante'].str.contains('Ferreira Filho') | partidas['tecnico_visitante'].str.contains('Ferreira Filho')]
partidas_filtro

Empty DataFrame
Columns: [partida_id, rodada, data, hora, mandante, visitante, tecnico_mandante, tecnico_visitante, vencedor, arena, mandante_placar, visitante_placar, mandante_estado, visitante_estado]
Index: []

In [31]:
partidas_filtro = partidas[partidas['tecnico_mandante'].str.contains('Soares Jardine') | partidas['tecnico_visitante'].str.contains('Soares Jardine')]
partidas_filtro

partida_id  rodada       data      hora     mandante    visitante  \
563         5551      19 2016-08-07  16:15:00   Santa Cruz    Sao Paulo   
575         5564      20 2016-08-14  16:18:00    Sao Paulo  Botafogo-RJ   
1477        6464      34 2018-11-15  18:59:00    Sao Paulo       Gremio   
1486        6473      35 2018-11-18  19:00:00    Sao Paulo     Cruzeiro   
1495        6482      36 2018-11-22  20:00:00        Vasco    Sao Paulo   
1508        6495      37 2018-11-26  20:01:00    Sao Paulo        Sport   
1517        6504      38 2018-12-02  17:04:00  Chapecoense    Sao Paulo   

               tecnico_mandante     tecnico_visitante     vencedor  \
563                   M. Mendes     A. Soares Jardine    Sao Paulo   
575           A. Soares Jardine    J. Ribeiro Ventura  Botafogo-RJ   
1477          A. Soares Jardine         Renato Gaúcho            -   
1486          A. Soares Jardine  L. Venker de Menezes    Sao Paulo   
1495  A. Valentim do Carmo Neto     A. Soares Jardine        Vasco   
1508          A. Soares Jardine             M. Mendes            -   
1517     C. dos Santos Oliveira     A. Soares Jardine  Chapecoense   

                      arena  mandante_placar  visitante_placar  \
563       Estádio do Arruda                1                 2   
575                 Morumbi                0                 1   
1477                Morumbi                1                 1   
1486                Morumbi                1                 0   
1495  Estádio Vasco da Gama                2                 0   
1508                Morumbi                0                 0   
1517            Arena Condá                1                 0   

     mandante_estado visitante_estado  
563               PE               SP  
575               SP               RJ  
1477              SP               RS  
1486              SP               MG  
1495              RJ               SP  
1508              SP               PE  
1517              SC               SP

In [35]:
partidas.loc[263, 'tecnico_mandante'] = 'W. Jose Ferreira Filho'
partidas.loc[516, 'tecnico_mandante'] = 'A. Soares Jardine'
partidas.loc[1042, 'tecnico_visitante'] = 'E. Carlos Tannus Faro'

In [37]:
partidas.isnull().sum()

partida_id           0
rodada               0
data                 0
hora                 0
mandante             0
visitante            0
tecnico_mandante     0
tecnico_visitante    0
vencedor             0
arena                0
mandante_placar      0
visitante_placar     0
mandante_estado      0
visitante_estado     0
dtype: int64

In [38]:
# Valores Ausentes: gols
gols.isnull().sum()

partida_id        0
clube             0
atleta            0
minuto            0
tipo_de_gol    6242
dtype: int64